mockrxandroidble/src/main/java/com/polidea/rxandroidble/mockrxandroidble/RxBleClientMock.java

```diff
@@ -0,0 +1,127 @@
+package com.polidea.rxandroidble.mockrxandroidble;
+
+import android.bluetooth.BluetoothGattCharacteristic;
+import android.bluetooth.BluetoothGattDescriptor;
+import android.bluetooth.BluetoothGattService;
+import android.support.annotation.Nullable;
+
+import com.polidea.rxandroidble.RxBleClient;
+import com.polidea.rxandroidble.RxBleDevice;
+import com.polidea.rxandroidble.RxBleDeviceServices;
+import com.polidea.rxandroidble.RxBleScanResult;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import rx.Observable;
+
+public class RxBleClientMock implements RxBleClient {
+
+    public static class Builder {
+
+        private Integer rssi;
+        private String deviceName;
+        private String deviceMacAddress;
+        private byte[] scanRecord;
+
+        private RxBleDeviceServices rxBleDeviceServices;
+
+        public RxBleClientMock build() {
```

Comment:
`deviceName` is actually optional but rssi, mac address, scan record looks like mandatory parameters.